django/contrib/admindocs/utils.py

```diff
@@ -104,14 +104,17 @@ def create_reference_role(rolename, urlbase):
     def _role(name, rawtext, text, lineno, inliner, options=None, content=None):
         if options is None:
             options = {}
+        # handle view and template with mixed case
+        exclude_name = ["view", "template"]
+        text_ref = text if name in exclude_name else text.lower()
         node = docutils.nodes.reference(
             rawtext,
             text,
             refuri=(
                 urlbase
                 % (
                     inliner.document.settings.link_base,
-                    text.lower(),
+                    text_ref,
```

Comment:
We know the `rolename` before, so checking `name` for each call seems like a waste. Can we do this in advance? E.g.

```diff
diff --git a/django/contrib/admindocs/utils.py b/django/contrib/admindocs/utils.py
index 6edff502ec..30df96ded0 100644
--- a/django/contrib/admindocs/utils.py
+++ b/django/contrib/admindocs/utils.py
@@ -101,6 +101,8 @@ ROLES = {
 
 
 def create_reference_role(rolename, urlbase):
+    is_case_sensitive = rolename in ["view", "template"]
+
     def _role(name, rawtext, text, lineno, inliner, options=None, content=None):
         if options is None:
             options = {}
@@ -111,7 +113,7 @@ def create_reference_role(rolename, urlbase):
                 urlbase
                 % (
                     inliner.document.settings.link_base,
-                    text.lower(),
+                    text if is_case_sensitive else text.lower(),
                 )
             ),
             **options,
```

Reply:
sure, nice